Add a guide to help new students set up their development environment

helloworld12321 opened the pull request:
These paragraphs were moved into this repository from `buzz-about/DEVELOPMENT.md`, and edited to stand alone.

I didn't go too far into how to use `git` or `npm`—I figured that was outside the scope of this document :slightly_smiling_face:

helloworld12321 committed:
Add directions for setting up GitHub, VS Code, git, GitKraken, and Node 

These paragraphs were moved into this repository from buzz-about/DEVELOPMENT.md, and edited to stand alone.

helloworld12321 committed:
Add Markdown all in One to the recommended extensions

It serves the important function of keeping the Table of Contents up to date (even though you don't necessarily need all of its features to work on the repo.)

floogulinc commented on .vscode/extensions.json:
```diff
@@ -0,0 +1,6 @@
+{
+  "recommendations": [
+    // We use Markdown All in One to generate the table of contents.
```
Use the same comment syntax as we do in Buzz About

```suggestion
    // Markdown All in One
    // https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one
```

floogulinc commented on README.md:
```diff
@@ -1,2 +1,106 @@
+
+## Installing Visual Studio Code
+
+Next, in order to read and write the code, you're going to need a text editor. There are a lot of good options, but the one that we've been using is [Visual Studio Code][], particularly because of its [live share][Live share] feature.
```
```suggestion
Next, in order to read and write the code, you're going to need a text editor. There are a lot of good options, but the one that we've been using is [Visual Studio Code][], particularly because of its [Live Share][Live share] feature.
```

helloworld12321 replied:
Oh, is it a proper noun? Cool! :relaxed:

floogulinc commented on README.md:
```diff
@@ -1,2 +1,106 @@
+
+You'll want to join the [Minnesota Pollinators][] organization ([Prof. KK Lamberty][] can invite you).
+
+If you're a student at Morris, be sure to sign up for the [student developer pack][Student developer pack] as well!
```
```suggestion
If you're a student at Morris, be sure to sign up for the [GitHub Student Developer Pack][Student developer pack] as well!
```

floogulinc commented on README.md:
```diff
@@ -1,2 +1,106 @@
+
+### On a Mac
+
+If you're on a Mac, you can get `git` as part of the XCode command-line developer tools. Open up Terminal and type `git --version`:
```
```suggestion
If you're on a Mac, you can get `git` as part of the Xcode command-line developer tools. Open up Terminal and type `git --version`:
```

helloworld12321 replied:
Ooh, thanks for catching that!

floogulinc commented on README.md:
```diff
@@ -1,2 +1,106 @@
+
+## Installing Node.js
+
+At this point, you should have all of the code for a repository on your computer, but you still don't have a way to compile and run it. For that, you'll need to install Node.js, a JavaScript interpreter.
```
```suggestion
At this point, you should have all of the code for a repository on your computer, but you still don't have a way to compile and run it. For that, you'll need to install Node.js, a JavaScript runtime.
```

helloworld12321 committed:
Change the comment format in extensions.json

Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>

helloworld12321 committed:
Capitalize "Live Share" in README.md

Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>

helloworld12321 committed:
Make "GitHub Student Developer Pack" a proper noun in README.md

Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>

helloworld12321 committed:
Lowercase the "c" in "Xcode"

Thanks for catching that, Paul!

Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>

helloworld12321 committed:
Use the word "runtime" to refer to Node.js, not "interpreter"

Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>

ErikRauer reviewed (changes requested):
This looks really good so far. You might want to include a section about getting the recommended VSCode Extensions that we use.

kklamberty commented on README.md:
```diff
@@ -1,2 +1,106 @@
+
+## Installing GitKraken
+
+`git` by itself is very spare, so we've been using it through a program called [GitKraken][]. (This program has a nice chart showing all of the revisions over time.) You can use GitKraken for free, but if you're a Morris student, you have access to the pro version through GitHub's student developer pack.
```
```suggestion
`git` by itself is very spare, so we've been using it through a program called [GitKraken][]. (This program has a nice chart showing all of the revisions over time.) You can use GitKraken for free, but if you're a Morris student, you have access to the pro version through [GitHub Student Developer Pack][GitHub's Student Developer Pack].
```

kklamberty commented on README.md:
```diff
@@ -1,2 +1,106 @@
+
+![The GitHub page for Buzz About, with the "Code" dropdown menu open, showing the title "Clone with HTTPS" and the start of the URL][i:Cloning from GitHUb]
+
+Now, you should be able to see the whole history of the project!
```
```suggestion
Now, you should be able to see the whole history of the project! Generally speaking, you will want to create a branch for any feature or change you wish to contribute and make a pull request in GitHub when you think it is ready for review to be added to the main branch in the repository. You can check out any branch locally using GitKraken by double clicking it. 
```

helloworld12321 replied:
Ooh, thank you, I'll add that! (I think I'll make it its own paragraph, if that's all right :slightly_smiling_face:)

kklamberty reviewed (commented):
I made a couple of suggestions, but overall, I really like this guide. Thanks for making it!

helloworld12321 committed:
Add a paragraph about using pull requests to add new features 

As part of the instructions for using GitKraken.

This paragraph was written by Prof. KK Lamberty. (I changed the phrasing just a little from the original).

Co-authored-by: KK Lamberty <lamberty@morris.umn.edu>

helloworld12321 committed:
Hyperlink "GitHub Student Developer Pack" in the GitKraken section

Also, capitalize "GitHub Student Developer Pack" for consistency with how we referred to it earlier in the document.

Co-authored-by: KK Lamberty <lamberty@morris.umn.edu>

helloworld12321 committed:
Add a note about installing the recommended VS Code extensions

At Erik's suggestion. Thanks!

Also, we moved the VS Code section of the document down, so that you don't install VS Code until after you've set up GitKraken. (Otherwise, there aren't any files to open!)
